From 5c6e80ad493992d60ef5b63f05717cb76ad2c6c5 Mon Sep 17 00:00:00 2001
From: Arun Ravindran <arun.ravindran@intel.com>
Date: Mon, 19 Nov 2012 16:48:00 +0200
Subject: [PORT FROM MAIN] Telephony: TZ update when the country code is NOK

BZ: 33279

3GPP Conformance 26.7.6.1.1 is failing because the telephony
framework doesn't update NITZ timezeone correctly.

Steps to Reproduce: Execute 3GPP Conformance 26.7.6.1.1
1. UE receives MM INFORMATION containing "0 hour" as a timezone.
Expected Results: UE updates local timezone with GMT+0 value.
Actual Results: UE doesn't update local timezone and keeps
preceding value.

This patch integrates the TZ update when the check TZ is not OK
(Country Code NOK)

Change-Id: I1eb07325413da30f4cc68621f329fca52a5b1e2c
Orig-Change-Id: I231fefaec52f31d4d6ecb05564e1805b13854d61
Category: aosp improvement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Emmanuel Delaude <emmanuelx.delaude@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../telephony/gsm/GsmServiceStateTracker.java      |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java b/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
index 0b75d49..d1520ce 100644
--- a/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
+++ b/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
@@ -1572,6 +1572,7 @@ final class GsmServiceStateTracker extends ServiceStateTracker {
                 mZoneOffset  = tzOffset;
                 mZoneDst     = dst != 0;
                 mZoneTime    = c.getTimeInMillis();
+                zone = getNitzTimeZone(mZoneOffset, mZoneDst, mZoneTime);
             }
 
             if (zone != null) {
-- 
1.7.4.1

